pyproject.toml

```diff
@@ -25,7 +25,7 @@ dependencies = [
     "cloudpickle>=3.0.0",
     "fabric>=3.0.0",
     "torch>=2.0.0",
-    "numpy",  # torch.distributed depends on numpy
+    "numpy<2",  # torch.distributed depends on numpy
```

Comment:
I don't think we want to force this constraint? End-users will not be allowed to use `numpy>=2` and I think that's too strict. Does our package (or `torch.distributed`) fail with such versions?

Reply:
Yes, `torch.distributed` seems to require this on some Python versions. Very annoying that `torch.distributed` doesn't have it as a pypi dependency and that we have to list it ourselves...

Reply:
Sorry I deleted it, but let's investigate a fix?

Reply:
I think numpy >= 2.0 is compatible with torch >= 2.3.0 according to this table

https://github.com/numpy/numpy/issues/26191#issue-2219105340